core/src/main/java/com/linecorp/armeria/server/cors/AbstractCorsPolicyBuilder.java

```diff
@@ -0,0 +1,320 @@
+    }
+
+    /**
+     * Specifies the headers that should be returned in the CORS {@code 'Access-Control-Allow-Headers'}
+     * response header.
+     *
+     * <p>If a client specifies headers on the request, for example by calling:
+     * <pre>{@code
+     * xhr.setRequestHeader('My-Custom-Header', "SomeValue");
+     * }</pre>
+     * the server will receive the above header name in the 'Access-Control-Request-Headers' of the
+     * preflight request. The server will then decide if it allows this header to be sent for the
+     * real request (remember that a preflight is not the real request but a request asking the server
+     * if it allow a request).
+     *
+     * @param headers the headers to be added to
+     *                the preflight {@code 'Access-Control-Allow-Headers'} response header.
+     * @return {@code this} to support method chaining.
+     */
+    public B allowRequestHeaders(final CharSequence... headers) {
+        requireNonNull(headers, "headers");
+        for (int i = 0; i < headers.length; i++) {
+            if (headers[i] == null) {
+                throw new NullPointerException("headers[" + i + ']');
+            }
+        }
+        Arrays.stream(headers).map(HttpHeaderNames::of).forEach(allowedRequestHeaders::add);
+        return self();
+    }
+
+    /**
+     * Returns HTTP response headers that should be added to a CORS preflight response.
+     *
+     * <p>An intermediary like a load balancer might require that a CORS preflight request
+     * have certain headers set. This enables such headers to be added.
+     *
+     * @param name the name of the HTTP header.
+     * @param values the values for the HTTP header.
+     * @return {@code this} to support method chaining.
+     */
+    public B preflightResponseHeader(final CharSequence name, final Object... values) {
+        requireNonNull(name, "name");
+        requireNonNull(values, "values");
+        for (int i = 0; i < values.length; i++) {
+            if (values[i] == null) {
+                throw new NullPointerException("values[" + i + ']');
+            }
+        }
+        preflightResponseHeaders.put(HttpHeaderNames.of(name),
+                                     new ConstantValueSupplier(Arrays.asList(values)));
+        return self();
+    }
+
+    /**
+     * Returns HTTP response headers that should be added to a CORS preflight response.
+     *
+     * <p>An intermediary like a load balancer might require that a CORS preflight request
+     * have certain headers set. This enables such headers to be added.
+     *
+     * @param name the name of the HTTP header.
+     * @param values the values for the HTTP header.
+     * @param <T> the type of values that the Iterable contains.
+     * @return {@code this} to support method chaining.
+     */
+    public <T> B preflightResponseHeader(final CharSequence name, final Iterable<T> values) {
+        requireNonNull(name, "name");
+        requireNonNull(values, "values");
```

Comment:
I'm wondering if we have to check the `values` contains `null` as we do in the method above.

Reply:
I think we may have to check them. Thanks. 
Added the logic checking if `values` contains `null` and making a new `ArrayList` so that the constructor of `CorsPolicy` doesn't have to check if it is an instance of `Iterable`.

Reply:
I think we need to check them. Added it.
Thanks